modules/identity/src/Volo.Abp.Identity.Blazor/Pages/Identity/RoleManagement.razor.cs

```diff
@@ -26,6 +39,99 @@ public RoleManagement()
             UpdatePolicyName = IdentityPermissions.Roles.Update;
             DeletePolicyName = IdentityPermissions.Roles.Delete;
             ManagePermissionsPolicyName = IdentityPermissions.Roles.ManagePermissions;
+            Toolbar = new PageToolbar();
+        }
+
+        protected override ValueTask SetEntityActionsAsync()
+        {
+            EntityActions
+                .Get<RoleManagement>()
+                .AddRange(new EntityAction[]
+                {
+                    new EntityAction
+                    {
+                        Text = L["Edit"],
+                        RequiredPolicy = UpdatePolicyName,
+                        Clicked = async (data) =>
+                        {
+                            await OpenEditModalAsync(data.As<IdentityRoleDto>());
+                        }
+                    },
+                    new EntityAction
+                    { 
+                        Text = L["Permissions"],
+                        RequiredPolicy = ManagePermissionsPolicyName,
+                        Clicked = async (data) =>
+                        {
+                            await PermissionManagementModal.OpenAsync(PermissionProviderName,
+                                data.As<IdentityRoleDto>().Name);
+                        }
+                    },
+                    new EntityAction
+                    {
+                        Text = L["Delete"],
+                        RequiredPolicy = DeletePolicyName,
+                        Clicked = async (data) => await DeleteEntityAsync(data.As<IdentityRoleDto>()),
+                        ConfirmationMessage = (data) => GetDeleteConfirmationMessage(data.As<IdentityRoleDto>())
+                    }
+                });
+
+            return base.SetEntityActionsAsync();
+        }
+
+        protected override ValueTask SetTableColumnsAsync()
+        {
+            TableColumns
+                .Get<RoleManagement>()
+                .AddRange(new TableColumn[]
+                {
+                    new TableColumn
+                    {
+                        Title = L["Actions"],
+                        Actions = EntityActions.Get<RoleManagement>()
+                    },
+                    new TableColumn
+                    {
+                        Title = L["RoleName"],
+                        Data = nameof(IdentityRoleDto.Name),
+                        Render = (data) =>
+                        {
+                            return (builder) =>
```

Comment:
Creating standard columns like that is not useful. We should think alternative approach if possible.